expression/column.go

```diff
@@ -193,13 +193,13 @@ func (col *Column) GetType() *types.FieldType {
 }
 
 // Eval implements Expression interface.
-func (col *Column) Eval(row []types.Datum) (types.Datum, error) {
-	return row[col.Index], nil
+func (col *Column) Eval(row types.Row) (types.Datum, error) {
+	return row.(types.DatumnRow)[col.Index], nil
 }
 
 // EvalInt returns int representation of Column.
-func (col *Column) EvalInt(row []types.Datum, sc *variable.StatementContext) (int64, bool, error) {
-	val := &row[col.Index]
+func (col *Column) EvalInt(row types.Row, sc *variable.StatementContext) (int64, bool, error) {
+	val := &row.(types.DatumnRow)[col.Index]
```

Comment:
Will this kind of type assert be time consuming?

Reply:
It's just temporary change, this type assertion will be removed in the next PR.